app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -95,5 +99,19 @@ def shibboleth
       end
     end
   end
+
+  private
+
+  # choose_shib_attr
+  # Shibboleth may return semi-column seperated attributes.
+  # The function returns the first one if attributes_str is not nil, otherwise
+  # it returns "(blank)" as a placeholder
+  def choose_shib_attr(attributes_str)
```

Comment:
Does this mean some attributes come in as lists?  Like if a person has multiple majors or whatnot?